core/domain/exp_jobs_one_off.py

```diff
@@ -529,6 +529,80 @@ def reduce(key, values):
         yield (key, values)
 
 
+class MathExplorationsImageGenerationAuditJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that finds all the explorations with math rich text components and
+    creates a temporary storage model with all the information required for
+    generating math rich text component SVG images.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        size_of_math_svg = 0
+        largest_math_expression = ''
+        for state in exploration.states.values():
+            html_string = ''.join(state.get_all_html_content_strings())
+            size_of_math_svg_in_state = (
+                (html_validation_service.
+                 estimate_size_of_svg_for_math_expressions_in_html(
+                     html_string))[0])
+            largest_math_expression_in_state = (
+                (html_validation_service.
+                 estimate_size_of_svg_for_math_expressions_in_html(
+                     html_string))[1])
+            if (len(largest_math_expression_in_state) >
+                    len(largest_math_expression)):
+                largest_math_expression = largest_math_expression_in_state
+            size_of_math_svg = size_of_math_svg + size_of_math_svg_in_state
+        value_dict = {
+            'exploration_id': item.id,
+            'size_of_math_svgs': size_of_math_svg,
+            'largest_math_expression': largest_math_expression
+        }
+        yield ('Information about math-expressions in explorations', value_dict)
```

Comment:
Actually, we should yield only one key because we find some output variables by iterating over all the yielded `item.id`'s.
If we yield item.id as a key, then the reduce() method will be called multiple times and we can't find all the required variables.
Hope it's clear or am I missing something